graphql/appointment/mutations.ts

```diff
@@ -0,0 +1,227 @@
+    ) {
+        appointments.forEach((appointment) => (appointment.organizers = mergeOrganizersWithSessionUserId(appointment.organizers, context)));
+        await createAppointments(appointments);
+        return true;
+    }
+
+    @Mutation(() => Boolean)
+    @AuthorizedDeferred(Role.OWNER)
+    // hasAccess is called in hasAccessMatch
+    // eslint-disable-next-line lernfair-lint/graphql-deferred-auth
+    async appointmentMatchCreate(@Ctx() context: GraphQLContext, @Arg('appointment') appointment: AppointmentCreateMatchInput) {
+        await hasAccessMatch(context, appointment.matchId);
+
+        await createMatchAppointment(appointment);
+        return true;
+    }
+
+    @Mutation(() => Boolean)
+    @AuthorizedDeferred(Role.OWNER)
+    // hasAccess is called in hasAccessMatch
+    // eslint-disable-next-line lernfair-lint/graphql-deferred-auth
+    async appointmentsMatchCreate(
+        @Ctx() context: GraphQLContext,
+        @Arg('matchId') matchId: number,
+        @Arg('appointments', () => [AppointmentCreateMatchInput]) appointments: AppointmentCreateMatchInput[]
+    ) {
+        await hasAccessMatch(context, matchId);
+        createMatchAppointments(matchId, appointments);
+        return true;
+    }
+
+    @Mutation(() => Boolean)
+    @AuthorizedDeferred(Role.OWNER)
+    // hasAccess is called in hasAccessSubcourse
+    // eslint-disable-next-line lernfair-lint/graphql-deferred-auth
+    async appointmentGroupCreate(@Ctx() context: GraphQLContext, @Arg('appointment') appointment: AppointmentCreateGroupInput) {
+        await hasAccessSubcourse(context, appointment.subcourseId);
+        await createGroupAppointment(appointment);
+        return true;
+    }
+
+    @Mutation(() => Boolean)
+    @AuthorizedDeferred(Role.OWNER)
+    // hasAccess is called in hasAccessSubcourse
+    // eslint-disable-next-line lernfair-lint/graphql-deferred-auth
+    async appointmentsGroupCreate(
+        @Ctx() context: GraphQLContext,
+        @Arg('subcourseId') subcourseId: number,
+        @Arg('appointments', () => [AppointmentCreateGroupInput]) appointments: AppointmentCreateGroupInput[]
+    ) {
+        await hasAccessSubcourse(context, subcourseId);
+
+        await createGroupAppointments(subcourseId, appointments);
+        return true;
+    }
+
+    @Mutation(() => Boolean)
+    @AuthorizedDeferred(Role.ADMIN, Role.OWNER)
+    // hasAccess is called in hasAccessSubcourse
+    // eslint-disable-next-line lernfair-lint/graphql-deferred-auth
+    async appointmentsUpdate(
+        @Ctx() context: GraphQLContext,
+        @Arg('appointmentsToBeUpdated', () => [AppointmentUpdateInput]) appointmentsToBeUpdated: AppointmentUpdateInput[]
+    ) {
+        await Promise.all(
+            appointmentsToBeUpdated.map(async (a) => {
+                const appointment = await getLecture(a.id);
+                if (appointment.subcourseId) {
```

Comment:
`await hasAccess(context, 'Lecture', appointment)` ? Otherwise the access check is inconsistent between update and the others